pkg/services/govmomi/pci/device.go

```diff
@@ -76,18 +76,28 @@ func ConstructDeviceSpecs(pciDeviceSpecs []infrav1.PCIDeviceSpec) []types.BaseVi
 	return pciDevices
 }
 
-func createBackingInfo(spec infrav1.PCIDeviceSpec) *types.VirtualPCIPassthroughDynamicBackingInfo {
-	return &types.VirtualPCIPassthroughDynamicBackingInfo{
-		AllowedDevice: []types.VirtualPCIPassthroughAllowedDevice{
-			{
-				VendorId: *spec.VendorID,
-				DeviceId: *spec.DeviceID,
+func createBackingInfo(spec infrav1.PCIDeviceSpec) types.BaseVirtualDeviceBackingInfo {
+	if spec.DeviceID != nil && spec.VendorID != nil {
```

Comment:
Let's maybe turn it around to have less checks (just nitpicking / optics) .
```go
	if spec.VGPUProfile != "" {
		return &types.VirtualPCIPassthroughVmiopBackingInfo{
			Vgpu: spec.VGPUProfile,
		}
	}
	
	... same as before
```

Reply:
I think checking the pointers is technically more safe because an empty VGPUProfile string makes it so the VM cannot find a host, avoiding nil-pointer panic, but is essentially setting a default which is not ideal.
The validation hook makes this point obsolete though, so is that enough to not bother checking them?